app/controllers/shops_controller.rb

```diff
@@ -4,23 +4,27 @@ class ShopsController < ApplicationController
   before_action :check_shop_exist, only: [:new, :create]
 
   def show 
-    @products = @shop.products 
+    @products = Product.available.where(shop: @shop)
+    @products = @shop.products if @user == current_user
```

Comment:
This makes almost 2 duplicate queries.
Use pundit scope here to filter out product scope to show

Reply:
noted and updated